### ISSUE TITLE
[4.0] select installation language

### ISSUE THREAD

**Issue by ghost** [joomla/joomla-cms#24481]
Quicker way of selecting the installation language in the language list (move to top of list).

Options:
1. Auto-install the installation language into Joomla (it's not going to be correct 100% of the time as it's not guaranteed to be the right language - but probably is more often than not)
2. At minimum put the current installation language to the top of the list of languages so people can easily find and select it.

**Comment by infograf768** [comment]
Reminder: custom distros may already include another language than en-GB and its default installation language selected.

**Comment by wilsonge** [comment]
how does the existing installer work with that? does it grey out languages that are installed or?

**Comment by infograf768** [comment]
Nothing special in 3.x.
The localise.xml file defines a default install language, thus it is already selected.
Nothing is modified when getting to the install language list if the user goes further there during installation. I would have to test again to find if the choice is given or not for defined default languages between both already present languages. (not on desktop now).

**Comment by brianteeman** [comment]
Are there still language distributions? I thought they had all stopped.

**Comment by infograf768** [comment]
@wilsonge 

Example of the French custom distro:
[Joomla_394-Stable-Full_Package_French_v1.zip](https://github.com/joomla/joomla-cms/files/3044474/Joomla_394-Stable-Full_Package_French_v1.zip)

The custom distro contains fr-FR in the pack

This is what we get when we install:

1. The installation uses the `installation/localise.xml`
```
<?xml version="1.0" encoding="utf-8"?>
<localise version="3.9" client="installation">
	<forceLang>fr-FR</forceLang>
	<helpurl></helpurl>
	<debug>0</debug>
	<sampledata>sample_data_fr.sql</sampledata>
	<params/>
</localise>
```

2. This will pre-select the installation language:

![Screen Shot 2019-04-04 at 18 10 56](https://user-images.githubusercontent.com/869724/55571063-090fe380-5705-11e9-8b49-f025d5f56f78.png)

3. It pre-selects a sample data

![Screen Shot 2019-04-04 at 18 12 52](https://user-images.githubusercontent.com/869724/55571184-4aa08e80-5705-11e9-9d24-87110f0bf302.png)

4. When we get to installing languages, it does NOT propose fr-FR as possible install:
(One can evidently install any other language)

![Screen Shot 2019-04-04 at 18 13 46](https://user-images.githubusercontent.com/869724/55571274-820f3b00-5705-11e9-8478-f8fce0414c44.png)

5. fr-FR is proposed as default site and admin languages

![Screen Shot 2019-04-04 at 18 17 30](https://user-images.githubusercontent.com/869724/55571559-0497fa80-5706-11e9-9155-9bf693f58b07.png)

in the joomla.sql we can find:
_extensions table:
```
[...]
(11, 0, 'com_languages', 'component', 'com_languages', '', 1, 1, 1, 1, '', '{"administrator":"fr-FR","site":"fr-FR"}', '', '', 0, '0000-00-00 00:00:00', 0, 0),
[...]
(600, 802, 'English (en-GB)', 'language', 'en-GB', '', 0, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
(601, 802, 'English (en-GB)', 'language', 'en-GB', '', 1, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
(602, 803, 'French (fr-FR)' , 'language', 'fr-FR', '', 0, 1, 0, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
(603, 803, 'French (fr-FR)' , 'language', 'fr-FR', '', 1, 1, 0, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
(700, 0, 'files_joomla', 'file', 'joomla', '', 0, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
(802, 0, 'English (en-GB) Language Pack', 'package', 'pkg_en-GB', '', 0, 1, 1, 1, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0),
(803, 0, 'French_(fr-FR) Language Pack' , 'package', 'pkg_fr-FR', '', 0, 1, 1, 0, '', '', '', '', 0, '0000-00-00 00:00:00', 0, 0);
[...]
```
_languages table
```
INSERT INTO `#__languages` (`lang_id`, `lang_code`, `title`, `title_native`, `sef`, `image`, `description`, `metakey`, `metadesc`, `sitename`, `published`, `access`, `ordering`) VALUES
(1, 'en-GB', 'English (en-GB)', 'English (United Kingdom)', 'en', 'en_gb', '', '', '', '', 1, 1, 2),
(2, 'fr-FR', 'Français (fr-FR)', 'Français (France)'      , 'fr', 'fr_fr', '', '', '', '', 1, 1, 1);
```

_menu types
```
INSERT IGNORE INTO `#__menu_types` (`id`, `asset_id`, `menutype`, `title`, `description`, `client_id`) VALUES
(1, 0, 'mainmenu', 'Menu principal', 'Le menu principal du site', 0);
```
The _usergroups is also customised
```
INSERT INTO `#__usergroups` (`id`, `parent_id`, `lft`, `rgt`, `title`) VALUES
(1, 0, 1, 18, 'Public'),
(2, 1, 8, 15, 'Enregistré'),
(3, 2, 9, 14, 'Auteur'),
(4, 3, 10, 13, 'Rédacteur'),
(5, 4, 11, 12, 'Validateur'),
(6, 1, 4, 7, 'Gestionnaire'),
(7, 6, 5, 6, 'Administrateur'),
(8, 1, 16, 17, 'Super Utilisateur'),
(9, 1, 2, 3, 'Invité');
```

And as well the _viewlevels
```
INSERT INTO `#__viewlevels` (`id`, `title`, `ordering`, `rules`) VALUES
(1, 'Accès public', 0, '[1]'),
(2, 'Accès enregistré', 2, '[6,2,8]'),
(3, 'Accès spécial', 3, '[6,3,8]'),
(5, 'Accès invité', 1, '[9]'),
(6, 'Accès super utilisateur', 4, '[8]');
```

**Comment by Quy** [comment]
Duplicate #22907?

**Comment by wilsonge** [comment]
Yup. Closed that one in favour of this as we have the discussion about the localisation packs there. However for anyone looking here make sure you look at #22907 as there are some nice design prototypes

**Comment by alikon** [comment]
Closing this due to not receiving required information to determine if this is a bug or not. If you feel this still needs review, please open a new tracker entry with as much information as possible to ensure it can be reviewed properly

**Comment by brianteeman** [comment]
I am re-opening this - it should not be closed. Please Please stop closing issues without reading them. And please stop using a default text that has no relevance to the topic.
